### PR TITLE
feat(lsp): support API for config file

### PR DESCRIPTION
**Pull request by kitsonk**
Closes: #13910

This add the `denoConfigTasks` experimental capability which handles a `deno/task` request from the client, which responds with an array of tasks from the current configuration file.

Clients need to implement surfacing them to the user, such as with denoland/vscode_deno#641.

### CODE REVIEW

File: cli/tests/integration/lsp_tests.rs [hunk 1]
```diff
@@ -579,6 +579,51 @@ fn lsp_import_map_config_file() {
...
+  assert!(maybe_err.is_none());
+  assert_eq!(
+    maybe_res,
+    Some(json!([
+      {
+        "name": "build",
+        "detail": "deno test"
+      },
+      {
+        "name": "some%test",
```
Comment (bartlomieju) [inline cli/tests/integration/lsp_tests.rs:620]: I think this is not right, we don't support `%` in task name - only alphanumeric, colons, underscores and dashes

Comment (kitsonk) [reply cli/tests/integration/lsp_tests.rs]: I just grabbed another fixture from another test without reading the test.  Interesting that it doesn't error. Parsed the tasks perfectly fine.

**Review by bartlomieju — APPROVED** [review]
LGTM

File: cli/tests/integration/lsp_tests.rs (hunk 1 above)
Comment (bartlomieju) [reply cli/tests/integration/lsp_tests.rs]: It parses fine, but @dsherret and I decided to play conservative for now and disallow most characters.

**Pull request merged**